src/Common/src/Internal/Cryptography/BasicSymmetricCipher.cs

```diff
@@ -33,14 +33,22 @@ protected BasicSymmetricCipher(byte[] iv, int blockSizeInBytes)
 
         public int BlockSizeInBytes { get; private set; }
 
-        public virtual void Dispose()
+        public void Dispose()
         {
-            if (IV != null)
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
That is the idea.  I was asking mainly because this whole hierarchy is internal.  It's not a big deal; given the computation performed by these objects, this isn't going to be a measurable cost.